src/mlpack/methods/ann/activation_functions/swish_function.hpp

```diff
@@ -0,0 +1,88 @@
+/**
+ * @file swish_function.hpp
+ * @author Vivek Pal
+ *
+ * Definition and implementation of the Swish function as described by
+ * Prajit Ramachandran, Barret Zoph & Quoc V. Le
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_SWISH_FUNCTION_HPP
+#define MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_SWISH_FUNCTION_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The swish function, defined by
+ *
+ * @f{eqnarray*}{
+ * f(x) &=& x \cdot \sigma(x) \\
+ * f'(x) &=& f(x) + \sigma(x) (1 - f(x)) \\
+ * \sigma(x) &=& frac{1}{1 + e^{-x}}
+ * @f}
+ */
+class SwishFunction
+{
+ public:
+  /**
+   * Computes the swish function.
+   *
+   * @param x Input data.
+   * @return f(x).
+   */
+  static double Fn(const double x)
+  {
+    return x / (1.0 + std::exp(-x));
+  }
+
+  /**
+   * Computes the swish function.
+   *
+   * @param x Input data.
+   * @param y The resulting output activation.
+   */
+  template<typename InputVecType, typename OutputVecType>
+  static void Fn(const InputVecType& x, OutputVecType& y)
+  {
+    y = x;
+
+    for (size_t i = 0; i < x.n_elem; i++)
+      y(i) = Fn(x(i));
```

Comment:
Do you think you could use armadillo expressions to write the function for a matrix type (e.g. by using ``arma::exp``)? This way it could be parallelised.

Reply:
Yes, sure thing. Will add the function for computing swish function using a matrix as input.